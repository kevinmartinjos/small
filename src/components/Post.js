import React from 'react';
import PostContentContainer from './PostContentContainer';

var PropTypes = React.PropTypes;

var Post = React.createClass({
	propTypes: {
		title: PropTypes.string.isRequired,
		content: PropTypes.string.isRequired,
		handleSelection: PropTypes.func
	},
	render() {
		return(
			<div className="Post">
				<h2 className="title">{this.props.title}</h2>
				<div onMouseUp={this.props.handleSelection}>
					<PostContentContainer content={this.props.content} />
				</div>
			</div>
		);
	}
});

export default Post;
