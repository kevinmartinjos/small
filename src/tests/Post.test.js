import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Post from '../components/Post';
import ReactTestUtils from 'react-addons-test-utils';

test('Post should be rendered', () => {
	const component = renderer.create(
		<Post title="Test" content="I am post" />
	);
	let tree = component.toJSON();
  	expect(tree).toMatchSnapshot();
});

test('Post should render html', () => {
	let component = ReactTestUtils.renderIntoDocument(<Post title="Test" content="I am a <span>span</span>" />);
	let node = ReactTestUtils.findRenderedDOMComponentWithTag(component, 'section');
	expect(node.textContent).toBe('I am a span');
});