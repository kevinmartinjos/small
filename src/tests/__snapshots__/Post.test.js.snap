exports[`test Post should be rendered 1`] = `
<div
  className="Post">
  <h2
    className="title">
    Test
  </h2>
  <div
    onMouseUp={undefined}>
    <div
      className="PostContentContainer">
      <section>
        I am post
      </section>
    </div>
  </div>
</div>
`;
